man/systemd.unit.xml

```diff
@@ -104,16 +104,24 @@
     <citerefentry><refentrytitle>systemd.scope</refentrytitle><manvolnum>5</manvolnum></citerefentry>.
     </para>
 
-    <para>Unit files are loaded from a set of paths determined during
-    compilation, described in the next section.</para>
-
-    <para>Unit files can be parameterized by a single argument called the "instance name". The unit
-    is then constructed based on a "template file" which serves as the definition of multiple
-    services or other units. A template unit must have a single <literal>@</literal> at the end of
-    the name (right before the type suffix). The name of the full unit is formed by inserting the
-    instance name between <literal>@</literal> and the unit type suffix. In the unit file itself,
-    the instance parameter may be referred to using <literal>%i</literal> and other specifiers, see
-    below.</para>
+    <para>Unit files are loaded from a set of paths determined during compilation, described in the next
+    section.</para>
+
+    <para>Valid unit names consist of a "name prefix" and a dot and a suffix specifying the unit type. The
+    "unit prefix" must consist of one or more valid characters (ASCII letters, digits, <literal>:</literal>,
+    <literal>-</literal>, <literal>_</literal>, <literal>.</literal>, and <literal>\</literal>). The total
+    length of the unit name including the suffix must not exceed 256 characters. The type suffix must be one
+    of <literal>.service</literal>, <literal>.socket</literal>, <literal>.device</literal>,
+    <literal>.mount</literal>, <literal>.automount</literal>, <literal>.swap</literal>,
+    <literal>.target</literal>, <literal>.path</literal>, <literal>.timer</literal>,
+    <literal>.slice</literal>, or <literal>.scope</literal>.</para>
```

Comment:
you sure of your list ? According to man:systemd.scope, there are no scope files... and I am not sure .device files make sense (they are valide, according to documentation) since all info should come from udev

Reply:
But this says "Valid unit names", not "Valid unit file names". I.e. `foo.scope` **is** a valid unit name, even though we wouldn't want a unit file. So I think this is OK. If you think that we should mention that for certain unit types files are not supported, please suggest a patch.